sql/inspect: add progress tracking to INSPECT

spilchen opened the pull request:
Previously, when you ran an INSPECT job we didn't maintain any kind of progress tracking. This commit adds it. The granularity that we use for progress tracking is with checks x spans. Note: each check won't always use all spans, so we only count spans that apply to the check.

The processors flow back the completed checks (per span) they have done. A goroutine runs at the coordinator that will periodically (every 10 seconds) update the job data with the progress update.

Informs #148297

Release note: none
Epic: CRDB-30356

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154136)
<!-- Reviewable:end -->

spilchen committed:
sql/inspect: add progress tracking to INSPECT

Previously, when you ran an INSPECT job we didn't maintain any kind of
progress tracking. This commit adds it. The granularity that we use for
progress tracking is with checks x spans. Note: each check won't always
use all spans, so we only count spans that apply to the check.

The processors flow back the completed checks (per span) they have done.
A goroutine runs at the coordinator that will peridocially (every 10
seconds) update the job data with the progress update.

Informs #148297

Release note: none
Epic: CRDB-30356

rafiss reviewed (approved):
lgtm! <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154136#-:-Oa6hnAE5-K3eUTsEV5H:b-ell9zw)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @kev-cao)



<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR!

bors r+
